File: Bowling/src/helge/Game.java
/*
 * Copyright (c) 2017. Helge Stenström
 */

package helge;

public class Game {
    private int rolls[] = new int[21];
    private int currentRoll = 0;

    public void roll(int pins) {
        rolls[currentRoll++] = pins;
    }

    public int score() {
        int score = 0;
        int frameIndex = 0;
        for (int frame=0; frame<10; frame++) {
            int frameScore =  rolls[frameIndex] + rolls[frameIndex+1];
            score += frameScore;
            if (frameScore == 10)
                score += rolls[frameIndex+2];
            frameIndex += 2;
        }
        return score;
    }
}

File: Python/Bowling/Bowling1/Game.py
class Game:
    def __init__(self):
        self.rolls = []

    def roll(self, pins):
        self.rolls += [pins]

    def getScore(self):
        totScore = 0
        rollNo = 0
        for frame in range(10):
            if self.rolls[rollNo] == 10:
                totScore += self.rolls[rollNo] + self.rolls[rollNo+1] + self.rolls[rollNo+2]
                rollNo += 1
            else:
                thisFrame = self.rolls[rollNo] + self.rolls[rollNo+1]
                totScore += thisFrame
                if thisFrame == 10:
                    totScore += self.rolls[rollNo + 2]
                rollNo += 2
        return totScore
